pkg/controller/cloud/node_controller.go

```diff
@@ -531,11 +535,11 @@ func ensureNodeExistsByProviderID(ctx context.Context, instances cloudprovider.I
 	return instances.InstanceExistsByProviderID(ctx, providerID)
 }
 
-func getNodeAddressesByProviderIDOrName(ctx context.Context, instances cloudprovider.Instances, node *v1.Node) ([]v1.NodeAddress, error) {
-	nodeAddresses, err := instances.NodeAddressesByProviderID(ctx, node.Spec.ProviderID)
```

Comment:
There are more places in this controller where we check the provider ID from the node object directly. I wonder if we should update all of those If we want to make this optimization? I think checking node.Spec.ProviderID is simpler but on first initialization clearly that field is not set yet so we can't use it. 

Could we instead add the provider ID modifier as a separate step prior to the other modifiers and set `Spec.ProviderID` to the local node object passed into getNodeModifiersFromCloudProvider?

Reply:
I can see this being a bad idea because we would make assumptions that the field is set in the API when we haven't committed it yet. @cheftako thoughts?

Reply:
> I can see this being a bad idea because we would make assumptions that the field is set in the API when we haven't committed it yet.

That's why I modified the func to accept `providerID, nodeName string` instead of `node v1.Node` (with the assumption that `.Spec.ProviderID` is set on the `node` arg).

Reply:
I think andrewsykim's point about wanting this fixed everywhere is legit, but I don't think it all has to be fixed in this PR. We should probably file a bug to do the follow on work.